crates/matrix-sdk/src/widget_api/permissions.rs

```diff
@@ -0,0 +1,24 @@
+//! Types and traits related to the permissions that a widget can request from a client.
+
+use async_trait::async_trait;
+
+/// A trait that must be implemented by an entity that performs the validation / authorization
+/// of the widget's permissions. Typically, it's a native client that "hosts" the widget.
```

Comment:
What's a "client" here? A Matrix client? And what significance does the "native" bit have?

Reply:
Perhaps poor wording on my side. I believe this trait will be implemented by a component that essentially provides the prompt to the user. "Native" is probably confusing here.

Should I reword it to something like: "A trait that must be implemented by a component that provides the functionality of deciding whether a widget is allowed to use certain capabilities (typically by providing a prompt to the user"?

Reply:
Sounds good, but you can leave out "A trait that must be implemented by", and please close the parentheses as well 😉

Reply:
Done!